src/drv_spi_hsoc_v1.c

```diff
@@ -138,6 +138,36 @@ static unsigned long spi_get_device_speed_representation(const FLASHER_SPI_CFG_T
 	return ulDevSpeed;
 }
 
+/** Convert a Spi Frequency in device speed representation into a kHz value.
+ * 
+ * @param ptCfg The configuration struct of the Spi Interface
+ * @param ulDeviceSpeed The device speed representation of the Spi Frequency
+ * 
+ * @return The Spi Frequency in kHz or 0 if no matching entry was found (error)
+*/
+static unsigned long spi_get_khz_speed_representation(const FLASHER_SPI_CFG_T* ptCfg, const unsigned long ulDeviceSpeed)
+{
+	unsigned long ulSpiClkSpeed;
+	unsigned int uiCnt;
+
+	ulSpiClkSpeed = ulDeviceSpeed;
```

Comment:
I would mask out everything except the speed bits:
ulSpiClkSpeed = ulDeviceSpeed & HOSTMSK(spi_control_register_CR_speed);